src/main/java/ch/abbts/nds/swe/swdt/starter/skill/SortingFactory.java

```diff
@@ -0,0 +1,40 @@
+package ch.abbts.nds.swe.swdt.starter.skill;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class SortingFactory {
+
+  Logger logger = LoggerFactory.getLogger(SortingFactory.class);
+
+  @Autowired
+  @Qualifier("ascendingStringOrderImpl")
+  private Sortable ascendingStrategy = new AscendingStringOrderImpl();
+
+  @Autowired
+  @Qualifier("descendingStringOrderImpl")
+  private Sortable descendingStrategy = new DescendingStringOrderImpl();
+
+  private List<Sortable> sortables;
```

Comment:
you can make the list final, so it will be always initialized in constructor.
otherwise you give people the possibility to have it as null, which can produce NPE in line 32.

If somebody will add another constructor and he will forget to init this, it will be null.